### ISSUE TITLE
strict ignores cleared transactions

### ISSUE THREAD

**Issue by taviso** [ledger/ledger#2189]
Is it intentional that `--strict` doesn't work with transactions marked cleared?

For example, I would expect this file to generate warnings with `--strict`:

```
2023/02/10 * Book Store
    Expenses:Entertainment:Books  $18
    Liabilities:Credit
```

But it doesn't:

```
$ ledger --args-only --strict -f test.ldg bal 
                 $18  Expenses:Entertainment:Books
                $-18  Liabilities:Credit
--------------------
                   0
```

If you don't mark it cleared, it works as expected:

```
line 3: Unknown account 'Expenses:Entertainment:Books'
```

**Comment by taviso** [comment]
I see this does seem to work as expected in the new release (3.3.0), so maybe this was fixed at some point!

**Comment by tbm** [comment]
There used to be an `--explicit` option, otherwise cleared would in fact mark everything in a txn as ok.

See:

commit 43b07fbab3b4c144eca4a771524e59c531ffa074
Author: CandyAngel <candyangel@electricjungle.org>
Date:   Tue Mar 31 12:47:53 2020 +0100

    Make explicit option a no-op
    
    Fixes #1877


NEWS.md:

- Entities are no longer regarded as defined due to being part of a cleared
  transaction. `--explicit` is effectively enabled by default and is now a
  no-op.
